docs/en/development/data_structures.md

```diff
@@ -5,4 +5,879 @@ slug: /internals
+```
+
+#### 3.1.11 Xattr
+
+Records extended attributes (Key-Value pairs) of a file, as follows
+
+```
+inode, key -> value
+```
+
+#### 3.1.12 Flock
+
+Records BSD locks (flock) of a file, specifically.
+
+```
+inode, sid, owner -> ltype
+```
+
+where sid is the client session ID, owner is a string of numbers, usually associated with a process, and ltype is the lock type, which can be 'R' or 'W'.
+
+#### 3.1.13 Plock
+
+Record POSIX record locks (fcntl) of a file, specifically
+
+```
+inode, sid, owner -> []plockRecord
+```
+
+Here plock is a more fine-grained lock that can lock only a certain segment of the file.
+
+```go
+type plockRecord struct {
+	ltype uint32 // lock type
+	pid   uint32 // process ID
+	start uint64 // start position of the lock
+	end   uint64 // end position of the lock
+}
+```
+
+This structure is encoded and stored in binary format, taking up 24 bytes.
+
+#### 3.1.14 DelFiles
+
+Records the list of files to be cleaned. Since data cleanup of files is an asynchronous and potentially time-consuming operation that can be interrupted by other factors, it is tracked by this list.
+
+```
+inode, length -> expire
+```
+
+where length is the length of the file and expire is the time when the file was deleted.
+
+#### 3.1.15 DelSlices
+
+Records delayed deleted Slices. When the Trash feature is enabled, old Slices deleted by the Slice Compaction are kept for the same amount of time as the Trash configuration, to be available for data recovery if necessary.
+
+```
+sliceId, deleted -> []slice
+```
+
+where sliceId is the ID of the new slice after compaction, deleted is the timestamp of the compaction, and the mapped value is the list of all old slices that were compacted, each slice encoding only the ID and size.
+
+```go
+type slice struct {
+	ID   uint64
+	Size uint32
+}
+```
+
+This structure is encoded and stored in binary format, taking up 12 bytes.
+
+#### 3.1.16 Sustained
+
+Records the list of files that need to be kept temporarily during the session. If a file is still open when it is deleted, the data cannot be cleaned up immediately, but needs to be held temporarily until the file is closed.
+
+```
+sid -> []inode
+```
+
+where sid is the session ID and the mapped value is the list of temporarily undeleted file inodes.
+
+### 3.2 Redis
+
+The common format of keys in Redis is `${prefix}${JFSKey}`, where
+
+- In standalone mode the prefix is an empty string, while in cluster mode it is a database number enclosed in curly braces, e.g. "{10}"
+- JFSKey is the Key of different data structures in JuiceFS, which are listed in the subsequent subsections
+
+In Redis Keys, integers (including inode numbers) are represented as decimal strings if not otherwise specified.
+
+#### 3.2.1 Setting
+
+- Key: `setting`
+- Value Type: String
+- Value: file system formatting information in JSON format
+
+#### 3.2.2 Counter
+
+- Key: counter name
+- Value Type: String
+- Value: value of the counter, which is actually an integer
```

Comment:
```suggestion
- Value: value of the counter, an integer
```

Reply:
need `actually`, cuz its type is String, while the value can be converted to an int